Improve logger warning

starkillerOG opened the pull request:
Include the sound mode raw that was not able to be matched in the logger warning if the sound mode could not be matched.

starkillerOG committed:
Improve logger warning

Include the sound mode raw that was not able to be matched in the logger warning if the sound mode could not be matched.

starkillerOG commented:
Just a verry small improvement to the logger warning text if the sound mode could not be matched.

starkillerOG commented:
@scarface-4711 would be helpfull to merge this to easily see which RAW_sound_mode schould be added to the mapping dict.

starkillerOG committed:
Do not show duplicate warnings

starkillerOG committed:
remove semicolon

starkillerOG committed:
Add "DTS Neural:X" to sound mode list

starkillerOG committed:
line to long

starkillerOG commented:
@scarface-4711 could you please merge this.
It will fix the multitude of warnings people are experiancing: [issue in HomeAssistant.](https://github.com/home-assistant/home-assistant/issues/15590#issuecomment-407381137)

It will incorperate the probelamtic sound_mode_raw in the error message so we can help people quicker.

It will add "DTS Nearal:X" to the sound mode mapping: [issue in denonavr](https://github.com/scarface-4711/denonavr/issues/57).

ol-iver commented on denonavr/denonavr.py:
```diff
@@ -1441,8 +1443,10 @@ def match_sound_mode(self, sound_mode_raw):
             pass
-        _LOGGER.warning("Not able to match sound mode, "
-                        "returning raw sound mode.")
+        if sound_mode_raw not in self._error_list:
```
Please add sound_mode_raw to self._sm_match_dict in case of a KeyError instead of creating the self._error_list attribute.

ol-iver replied:
with a mapping to itself

ol-iver commented:
@starkillerOG  I'm on a business trip right now. Can merge this on the weekend.

starkillerOG committed:
remove error_list

remove error_list and use the sound_mode_dict and sm_match_dict instead.
enforce uppder case in construct sm_match_dict, just in case.

starkillerOG committed:
line to long

starkillerOG committed:
add 'STANDARD(DOLBY)' to sound mode mapping

starkillerOG committed:
Add 'Dolby Surround' to sound mode mapping

ol-iver commented:
@starkillerOG: version 0.7.5 is available

starkillerOG commented:
Thanks, just made a PR in Homeassistant.
